pkg/ccl/backupccl/backup_test.go

```diff
@@ -3616,6 +3616,174 @@ func TestBackupRestoreSequence(t *testing.T) {
+			}
+
+			newDB.ExpectErr(t, `pq: cannot restore sequence "seq" without referenced owner table`,
+				`RESTORE TABLE seq FROM $1`, backupLoc)
+			newDB.Exec(t, `RESTORE TABLE seq FROM $1 WITH skip_missing_sequence_owners`, backupLoc)
+
+			seqDesc := sqlbase.TestingGetTableDescriptor(kvDB, keys.SystemSQLCodec, "d", "seq")
+			if seqDesc.SequenceOpts.HasOwner() {
+				t.Fatal("unexpected sequence owner after restore")
+			}
+		})
+
+	t.Run("test restoring database should preserve ownership dependency", func(t *testing.T) {
+		tc := testcluster.StartTestCluster(t, singleNode, base.TestClusterArgs{ServerArgs: args})
+		defer tc.Stopper().Stop(context.Background())
+
+		newDB := sqlutils.MakeSQLRunner(tc.Conns[0])
+		kvDB := tc.Server(0).DB()
+
+		newDB.Exec(t, `RESTORE DATABASE d FROM $1`, backupLoc)
+
+		tableDesc := sqlbase.TestingGetTableDescriptor(kvDB, keys.SystemSQLCodec, "d", "t")
+		seqDesc := sqlbase.TestingGetTableDescriptor(kvDB, keys.SystemSQLCodec, "d", "seq")
+
+		if !seqDesc.SequenceOpts.HasOwner() ||
+			seqDesc.SequenceOpts.SequenceOwner.OwnerTableID != tableDesc.ID ||
+			seqDesc.SequenceOpts.SequenceOwner.OwnerColumnID != tableDesc.GetColumns()[0].ID {
+			t.Fatal("unexpected/no sequence owner after restore")
+		}
+
+		if len(tableDesc.GetColumns()[0].OwnsSequenceIds) != 1 ||
+			tableDesc.GetColumns()[0].OwnsSequenceIds[0] != seqDesc.ID {
+			t.Fatal("unexpected/no sequence owned by column")
+		}
+	})
+
+	t.Run("test restoring two databases removes cross-database ownership dependency",
+		func(t *testing.T) {
+			tc := testcluster.StartTestCluster(t, singleNode, base.TestClusterArgs{ServerArgs: args})
+			defer tc.Stopper().Stop(context.Background())
+
+			newDB := sqlutils.MakeSQLRunner(tc.Conns[0])
+			kvDB := tc.Server(0).DB()
+
+			newDB.ExpectErr(t, `pq: cannot restore`,
+				`RESTORE DATABASE d2 FROM $1`, backupLocD2)
+			newDB.Exec(t, `RESTORE DATABASE d2 FROM $1 WITH skip_missing_sequence_owners`, backupLocD2)
+
+			tableDesc := sqlbase.TestingGetTableDescriptor(kvDB, keys.SystemSQLCodec, "d2", "t")
+			numOwnedSequences := len(tableDesc.GetColumns()[0].OwnsSequenceIds)
+			if numOwnedSequences != 0 {
+				t.Fatalf("expected restored table to own 0 sequences. got: %v", numOwnedSequences)
+			}
+			newDB.ExpectErr(t, `pq: cannot restore`,
+				`RESTORE DATABASE d3 FROM $1`, backupLocD3)
+			newDB.Exec(t, `RESTORE DATABASE d3 FROM $1 WITH skip_missing_sequence_owners`, backupLocD3)
+
+			seqDesc := sqlbase.TestingGetTableDescriptor(kvDB, keys.SystemSQLCodec, "d3", "seq")
+			if seqDesc.SequenceOpts.HasOwner() {
+				t.Fatal("unexpected sequence owner after restore")
+			}
+
+			// Sequence dependencies inside the database should still be preserved.
+			sd := sqlbase.TestingGetTableDescriptor(kvDB, keys.SystemSQLCodec, "d3", "seq2")
+			td := sqlbase.TestingGetTableDescriptor(kvDB, keys.SystemSQLCodec, "d3", "t")
+			if !sd.SequenceOpts.HasOwner() || sd.SequenceOpts.SequenceOwner.OwnerTableID != td.ID {
```

Comment:
Maybe we could do:
```
require.True(t, sd.SequenceOpts.HasOwner())
require.Equal(t, td.ID, sd.SequenceOpts.SequenceOwner.OwnerTableID)
```
so that if this test fails, it's a bit clearer what is going wrong (same elsewhere in the tests).